crates/ruff_python_semantic/src/all.rs

```diff
@@ -66,34 +68,78 @@ impl Ranged for DunderAllDefinition<'_> {
     }
 }
 
-/// Extract the names bound to a given __all__ assignment.
-///
-/// Accepts a closure that determines whether a given name (e.g., `"list"`) is a Python builtin.
-pub fn extract_all_names<F>(stmt: &Stmt, is_builtin: F) -> (Vec<DunderAllName>, DunderAllFlags)
-where
-    F: Fn(&str) -> bool,
-{
-    fn add_to_names<'a>(
-        elts: &'a [Expr],
-        names: &mut Vec<DunderAllName<'a>>,
-        flags: &mut DunderAllFlags,
-    ) {
-        for elt in elts {
-            if let Expr::StringLiteral(ast::ExprStringLiteral { value, range }) = elt {
-                names.push(DunderAllName {
-                    name: value.to_str(),
-                    range: *range,
-                });
+impl<'a> SemanticModel<'a> {
```

Comment:
@MichaReiser how idiomatic/unidiomatic is it to have this `impl` block here rather than in `model.rs`, would you say? I think it _is_ quite nice to have all the functionality related to `__all__` in a separate module, but I'm not sure how common this kind of thing is